include/exadg/operators/operator_base.cpp

```diff
@@ -637,6 +643,11 @@ OperatorBase<dim, Number, n_components>::apply_add_block_diagonal_elementwise(
     unsigned int const n_faces = dealii::ReferenceCells::template get_hypercube<dim>().n_faces();
     for(unsigned int face = 0; face < n_faces; ++face)
     {
+      AssertThrow(matrix_free->get_dof_handler(data.dof_index)
+                    .get_triangulation()
+                    .all_reference_cells_are_hyper_cube(),
+                  dealii::ExcMessage("Can't do cell based loop over faces for simplices."));
```

Comment:
I think we should not place these 3 Asserts inside the innermost loops over the faces (for performance reasons). Why not in `apply_inverse_block_diagonal()` where you have added the other assert.

Reply:
I wanted to add the Assert deeper in code. I didn't know it could lead to worse performance.